infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InitializedState.java

```diff
@@ -77,8 +77,17 @@ public void onEntry(LogReplicationState from) {
 
     @Override
     public void onExit(LogReplicationState to) {
+        LogReplicationStateType type = to.getType();
+        LogReplicationMetadata.SyncType toSyncType = null;
+
+        if (type.equals(LogReplicationStateType.IN_SNAPSHOT_SYNC) || type.equals(LogReplicationStateType.WAIT_SNAPSHOT_APPLY)) {
+            toSyncType = LogReplicationMetadata.SyncType.SNAPSHOT;
+        } else if (type.equals(LogReplicationStateType.IN_LOG_ENTRY_SYNC)) {
+            toSyncType = LogReplicationMetadata.SyncType.LOG_ENTRY;
+        }
+
         if (to != this || to.getType() != LogReplicationStateType.ERROR) {
-            fsm.getAckReader().startSyncStatusUpdatePeriodicTask();
+            fsm.getAckReader().startSyncStatusUpdatePeriodicTask(toSyncType);
```

Comment:
From looking at the code, something is still not correct. On first execution ever, this will set the status to ongoing (onExit) i.e. before the InSnapshot or InLogEntry sync has started. But take the example of InSnapshotSyncState, we would set it to ONGOING, but the internal SnapshotSyncInfo won't be correctly populated cause you don't have the info at this level, it will be correctly populated on InSnapshotSync entry (which should happen right away unless the node crashes). I think then the correct thing is either to set it to NOT_STARTED, or not make the periodic task thread update the status before the first initialization from the state.

Reply:
Is this the same issue PR #3435 trying to resolve?